src/main/java/org/broadinstitute/consent/http/service/UserService.java

```diff
@@ -454,12 +454,135 @@ public void hasValidActiveERACredentials(User user) {
     }
   }
 
+  /**
+   * Compliance method defined in
+   * <a href="https://broadworkbench.atlassian.net/browse/DT-1607">this ticket</a> that implements
+   * a set of rules in order to ensure Library Card and Institution matching rules are
+   * adhered to when authorizing users of the system.
+   * @param user the User being evaluated
+   * @return user with the Institution and Library Card rules applied
+   */
+  public User enforceInstitutionAndLibraryCardRules(User user) {
+    Institution institutionFromEmail = institutionService.findInstitutionForEmail(user.getEmail());
+    Institution institutionFromDatabase = null;
+    try {
+      institutionFromDatabase = institutionService.findInstitutionById(user.getInstitutionId());
+    } catch (NotFoundException nfe) {
+      // do nothing.
+    }
```

Comment:
Why do we need `institutionFromDatabase` here? Its only use is to get `institutionId`, which we already have, and to set `institution` in `User`, which is unused since any changes to `User` result in re-reading it from the database below.